src/Core/Content/Product/Helper/SalesChannelProductHelper.php

```diff
@@ -0,0 +1,100 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Product\Helper;
+
+use Shopware\Core\Content\Product\SalesChannel\SalesChannelProductEntity;
+use Shopware\Core\Content\Property\Aggregate\PropertyGroupOption\PropertyGroupOptionCollection;
+use Shopware\Core\Content\Property\PropertyGroupCollection;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+
+class SalesChannelProductHelper
+{
+    private SystemConfigService $systemConfigService;
+
+    public function __construct(
+        SystemConfigService $systemConfigService
+    ) {
+        $this->systemConfigService = $systemConfigService;
+    }
+
+    public function sortProperties(SalesChannelProductEntity $product): PropertyGroupCollection
```

Comment:
```suggestion
    public function build(SalesChannelProductEntity $product): SalesChannelProductEntity
```

do the logic of the others methods in this method.

Reply:
Additionally add an abstract class so this can easily be extended